### PR TITLE
[juce] Fix cross builds, cleanup

### PR DESCRIPTION
**Pull request by dg0yt**
For https://github.com/microsoft/vcpkg/pull/42593#issuecomment-2529328752.

This PR focuses on improving cross builds (in particular in vcpkg CI) and on improving maintainability.

In its current form, building most of the port's binary artifacts is delegated to the consuming projects, via CMake functions. This includes vendored copies of harfbuzz, zlib etc. Changing this aspect is out of scope for this PR, but should be addressed in the future.

### CODE REVIEW

**Review by aminya — CHANGES_REQUESTED** [review]

File: ports/juce/portfile.cmake [hunk 1]
```diff
@@ -112,44 +125,29 @@ vcpkg_cmake_config_fixup()
...
+if(tool_names)
+  vcpkg_copy_tools(TOOL_NAMES ${tool_names} AUTO_CLEAN)
 endif()
 
-# Copy JUCE modules including the cpp/cmake files
-file(GLOB JUCE_MODULES_FOLDERS "${CURRENT_PACKAGES_DIR}/include/JUCE-${VERSION}/modules/*")
-foreach(JUCE_MODULE_FOLDER IN LISTS JUCE_MODULES_FOLDERS)
-  file(COPY "${JUCE_MODULE_FOLDER}" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
-endforeach()
-
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/juce/portfile.cmake:138]: The full context is https://github.com/microsoft/vcpkg/pull/42615/commits/a4f406d349af01ea6965514701d7fe48ca46ef76, i.e. there is also a patch, and I could see the right path in installed CMake files.
But juce probably deserves a test port.

TBH I would prefer to leave evertyhing only in `include/JUCE-${VERSION}`. Why did you add the copy in `include`? Is this packages usable without the CMake macros?

[the remark this answers is not in the document]
Comment (aminya) [reply ports/juce/portfile.cmake:138]: SJuce doesn't work without the CMake macros. The includes are needed for the user code, and the modules are needed for the CMake macros. I can help you add a test if needed, but the easiest thing is just to duplicate the headers.

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/juce/portfile.cmake]: Includes at `include/JUCE-${VERSION}/modules` are sufficient.

**Comment by LilyWangLL** [comment]
All features passed with following triplets:
```
x64-windows
x64-windows-static
```
Usage test passed on `x64-windows`.

**Review by LilyWangLL — APPROVED** [review]
